core/src/main/java/org/jruby/ext/socket/RubyIPSocket.java

```diff
@@ -46,51 +48,37 @@
  * @author <a href="mailto:ola.bini@ki.se">Ola Bini</a>
  */
 @JRubyClass(name="IPSocket", parent="BasicSocket")
-public class RubyIPSocket extends RubyBasicSocket {
+public abstract class RubyIPSocket extends RubyBasicSocket {
```

Comment:
I presume this means you could extend IPSocket and make your own...not that I know how you'd do that with so much hidden in C.

I agree we should probably undo this change.